### PR TITLE
Remove unused code.

### PR DESCRIPTION
**Pull request by paulp**
This almost certainly removes some code which should be deprecated instead. My motivations in sending a PR are part that I want to see what breaks and part that I just want to throw it out there and see what happens, because the alternative is most likely that it dies in the womb.

### CODE REVIEW

**Comment by paulp** [comment]
Also, some might find it interesting to look at because of what it reveals to be unused.

**Comment by paulp** [comment]
Review @adriaanm

**Comment by paulp** [comment]
I guess that broke less than I thought. I put back four deprecated methods and it all passed.

File: src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala [hunk 1]
```diff
@@ -49,21 +49,6 @@ abstract class TreeBrowsers {
    * Java Swing pretty printer for Scala abstract syntax trees.
    */
   class SwingBrowser {
-
-    def browse(t: Tree): Tree = {
```
Comment (dragos) [inline src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala:53]: I found this useful when debugging a phase, either by manually calling it, or dropping in a REPL and calling it from there. Not saying it should stay, but it's not surprising there are no calls to it in the code base.

**Comment by dragos** [comment]
LGTM (and this PR marks the first useful feedback from the IDE validator, if I'm not mistaken!).

**Comment by retronym** [comment]
I imagine that many of the presently unused bits of bcode will become used in the upcoming commits. Nothing a rebase won't resurrect, but maybe we can defer the cull there instead.

File: src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala (hunk 1 above)
Comment (paulp) [inline src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala:53]: I try to view this sort of thing through the lens of "what are the chances that when someone is in need of similar code, they will find and benefit from this?" It being useful isn't enough if the chance of it being found is negligible, which it is in my experience if it isn't more deeply integrated.

**Comment by paulp** [comment]
@retronym I recognized that, but there are always an infinity of reasons why some piece of unused code might be more used in the future, and many is the slip twixt the cup and the lip. The burden of carrying the code should not fall on master. Maybe if you're talking about a pull request which is open and ready for merge, but I think anything less than that should be treated as "one possible future."

**Comment by som-snytt** [comment]
Git should track not only history but possible futures!  When you merge, you could get a conflict like "Conflict if paulp ever merges xxxx or retronym ever merges yyyy."

For that matter, too bad scalac isn't git-aware.  Instead of `SwingBrowser with DebugBrowsing`, you'd write `with SwingBrowser_abcd` to get the version you want.

**Comment by paulp** [comment]
> For that matter, too bad scalac isn't git-aware. Instead of SwingBrowser with DebugBrowsing, you'd write with SwingBrowser_abcd to get the version you want.

FYI such awareness is a key piece of my Thing To Be.

**Comment by som-snytt** [comment]
The future is now, but it is trapped inside paulp's brain.

What's more, a certain alternate future is buried deep in his subconscious.

Who can say which future is most desirable?

Wait, are you doing a ClassPath that pulls sources out of branches?  A Scala SourcePath would have to compile the selected commits against the remaining class path, then supply the classes that result.

**Comment by adriaanm** [comment]
As IDE validation builds (most of) SBT, I think this is good to go.
@jsuereth or @harrah, please correct me if not. I'd like to merge this before the end of the week.

**Comment by harrah** [comment]
LGTM.

**Comment by paulp** [comment]
I'll rebase it and then merge since it sounds like it LTG everyone, plus or minus.

**Pull request merged**